### ISSUE TITLE
Allow anonymization/deletion of personally identifiable information

### ISSUE THREAD

**Issue by SDavisMedia** [easydigitaldownloads/edd-checkout-fields#161]
## Enhancement Request
CFM will need to export and delete payment and customer data if requested in the WP Core Privacy tools.

### Justification or use case
This is a [GDPR](https://github.com/easydigitaldownloads/easy-digital-downloads/issues/6216) issue.

**Comment by SDavisMedia** [comment]
These are my testing scenarios and results.

#### Data saved as Payment Meta:

##### Export:
✅ **Data included in export** - All good. The data shows up in the Customer Orders section.
✅ **Data excluded from export** - All good. No data shows up in the Customer Orders section.

##### Erase:
✅ **Data set to `No Action` during erasure** - All good. CFM payment meta untouched regardless of what happens to the payment itself (unless the payment is deleted, obviously).
❌ **Data set to `Anonymize` on erasure** - No good. `_wp_old_slug` value isn't anonymized.
✅ **Data set to `Delete` during erasure** - All good. CFM payment meta deleted regardless of what happens to the payment itself.

#### Data saved as User Meta:

When CFM data is saved as User Meta, it is saved as `_usermeta` and does not touch the associated payment or customer record. While this data is displayed on the customer record in its own tab, that is only because it's pulled through the link between the customer record and the user record. That said, when a customer record is anonymized, it is detached from any WP user account, breaking the link that displays the aforementioned user meta saved through CFM.

In other words, the new Privacy Settings settings on the CFM fields only matter when data is saved as Payment Meta. If the data is saved as User Meta, it will remain there unless WordPress itself does something about it during an erase. 🤷‍♂️

**Comment by SDavisMedia** [comment]
Sorry:

**Data set to Anonymize on erasure** is ✅✅✅ ... fixed when tested on EDD `fixes/2.9.2`.

With that, in my opinion, this is good to go. It seems to all work as expected. :+1:
